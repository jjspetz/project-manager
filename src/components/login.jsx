import React, {Component} from 'react';
import {connect} from 'react-redux';
import {togglePopover, setProvider} from '../actions';
import {bindActionCreators} from 'redux';
import {auth} from '../fire.js';

import Popover from 'material-ui/Popover';
import Menu from 'material-ui/Menu';
import MenuItem from 'material-ui/MenuItem';
// import RaisedButton from 'material-ui/RaisedButton';

class DropdownLogin extends Component {

  handleClose = () => {
    this.props.togglePopover(!this.props.openLogin);
  };

  handleLoginChoice = (value) => {
    this.props.togglePopover(!this.props.openLogin);
    auth(value)
      .then(function (user) {
        console.log(user)
      })
      .catch(function (e) {
        console.log(e);
      });
  }

  render() {
      return (
          <Popover
            open={this.props.openLogin}
            anchorEl={this.props.anchorEl}
            anchorOrigin={{horizontal: 'left', vertical: 'bottom'}}
            targetOrigin={{horizontal: 'left', vertical: 'top'}}
            onRequestClose={this.handleClose}
          >
            <Menu>
              <MenuItem primaryText="Google" onTouchTap={() => this.handleLoginChoice('google')}/>
              <MenuItem primaryText="Github" onTouchTap={() => this.handleLoginChoice('github')}/>
              <MenuItem primaryText="Facebook" onTouchTap={() => this.handleLoginChoice('facebook')}/>
            </Menu>
          </Popover>
      );
    }
  }

function mapStateToProps(state) {
  return {
    openLogin: state.openLogin,
    anchorEl: state.anchorEl,

  };
}

function mapDispatchToProps(dispatch) {
  return bindActionCreators(
    {
      togglePopover,
    }, dispatch
  )
}
export default connect(mapStateToProps, mapDispatchToProps)(DropdownLogin);
