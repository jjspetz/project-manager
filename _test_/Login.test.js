// how to handle modules in component class
import React from 'react';
import { mount, shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import { DropdownLogin } from '../src/components/login.jsx';

const wrapper = shallow(<DropdownLogin />);

describe('<DropdownLogin />', () => {
  it('should render', () => {
    expect(wrapper).toBeTruthy();
  });
  it('should contain a div', () => {
    expect(wrapper.find('div')).toBeTruthy();
  });
  it('should match snapshot', () => {
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
