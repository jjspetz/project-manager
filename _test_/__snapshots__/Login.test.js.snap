// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DropdownLogin /> should match snapshot 1`] = `
<Popover
  anchorOrigin={
    Object {
      "horizontal": "left",
      "vertical": "bottom",
    }
  }
  animated={true}
  autoCloseWhenOffScreen={true}
  canAutoPosition={true}
  onRequestClose={[Function]}
  open={false}
  style={
    Object {
      "overflowY": "auto",
    }
  }
  targetOrigin={
    Object {
      "horizontal": "left",
      "vertical": "top",
    }
  }
  useLayerForClickAway={true}
  zDepth={1}
>
  <Menu
    autoWidth={true}
    desktop={false}
    disableAutoFocus={false}
    initiallyKeyboardFocused={false}
    maxHeight={null}
    multiple={false}
    onChange={[Function]}
    onEscKeyDown={[Function]}
    onItemTouchTap={[Function]}
    onKeyDown={[Function]}
  >
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      onTouchTap={[Function]}
      primaryText="Google"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      onTouchTap={[Function]}
      primaryText="Github"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
    <MenuItem
      anchorOrigin={
        Object {
          "horizontal": "right",
          "vertical": "top",
        }
      }
      checked={false}
      desktop={false}
      disabled={false}
      focusState="none"
      insetChildren={false}
      onTouchTap={[Function]}
      primaryText="Facebook"
      targetOrigin={
        Object {
          "horizontal": "left",
          "vertical": "top",
        }
      }
    />
  </Menu>
</Popover>
`;
